doc/vim-iced.txt

```diff
@@ -217,6 +218,48 @@ SHADOW-CLJS~
 <
   To start CLJS REPL, see |vim-iced-shadow-cljs|.
 
+                                              *vim-iced-manual-rebel-readline*
+REBEL READLINE~
+
+  If you use Rebel Readline or want to embed an nREPL in your application,
+  then one approach is to create a `user.clj` with the following:
+>  
+  (ns user (:require [nrepl.server :as nrepl-server]
+                  [clojure.java.io :as io]))
+  
+  (def nrepl-port 7888)
+  (defonce nrepl-server (atom nil))
+
+  (defn cider-middleware
+    "Get cider middleware, see 
+    https://github.com/clojure-emacs/cider-nrepl/issues/447"
+    []
+    (require 'cider.nrepl)
+    (map resolve (var-get (ns-resolve 'cider.nrepl 'cider-middleware))))
```

Comment:
`cider-middleware` contains all middlewares provided from cider-nrepl.
https://github.com/clojure-emacs/cider-nrepl/blob/v0.25.4/src/cider/nrepl/middleware.clj#L7

Now `track-state` middleware may lead vim-iced's hang-up. (c.f. #170)
So, to be safe, we should specify the middleware separately, as in Leiningen's example.

Reply:
Btw, now that https://github.com/clojure-emacs/cider-nrepl/issues/447 is fixed, this `resolve` step is not necessary.

Reply:
Thank you for the information!

Reply:
@liquidz My poor understanding of Clojure and cider are showing here. I believe you wanted it to specifically load the cider middleware that you have for Leiningen. I have done this - probably in the ugliest way possible. 

I removed the resolve step as bbatsov suggested.